src/meadowrun/aws_integration/grid_tasks_sqs.py

```diff
@@ -389,15 +433,16 @@ async def receive_results(
     download_keys_received: Set[str] = set()
     wait = 4  # it's rare for any results to be ready in <4 seconds
     workers_exited_wait_count = 0
-    while not stop_receiving.is_set() and (workers_exited_wait_count < 3 or wait == 0):
-
+    while not stop_receiving.is_set() and workers_exited_wait_count < 3:
```

Comment:
I think it's good to have `wait == 0` in there? The idea was that even if the workers have exited, we should keep querying S3 until we have no new results showing up

Reply:
good catch, was unintentional after merge.